src/gui/http.go

```diff
@@ -74,22 +98,25 @@ func LaunchWebInterfaceHTTPS(host, staticDir string, daemon *daemon.Daemon, cert
 		return err
 	}
 
-	listener, err = tls.Listen("tcp", host, &tls.Config{Certificates: certs})
+	s.listener, err = tls.Listen("tcp", host, &tls.Config{Certificates: certs})
 	if err != nil {
 		return err
 	}
 
 	// Runs http.Serve() in a goroutine
-	serve(listener, NewGUIMux(appLoc, daemon), quit)
 	return nil
+
 }
 
-func serve(listener net.Listener, mux *http.ServeMux, q chan struct{}) {
+func (s *Server) Serve() {
+	mux := NewGUIMux(s.appLoc, s.daemon)
```

Comment:
The main thread calls serve in a goroutine, serve should not also run a goroutine